src/components/modals/EditDialogModal.tsx

```diff
@@ -241,21 +241,25 @@ class EditDialogModal extends React.Component<Props, ComponentState> {
             newTrainDialog.definitions = definitions
 
             // Content could come from button submit
-            const buttonSubmit = activity.value ? activity.value['submit'] : null
-            const userInput: CLM.UserInput = { text: buttonSubmit || activity.text! }
-            
+            const userInput: CLM.UserInput = { text: activity.text! }
+
             // Allow webchat to scroll to bottom 
             this.props.clearWebchatScrollPosition()
 
             // If there's an error when I try to continue, reset webchat to ignore new input
             await this.props.setErrorDismissCallback(this.resetWebchat)
 
-            // If button submit, and have selected action, insert at that step
-            if (this.state.selectedActivity) {
+            // For now always add button response to bottom of dialog even
+            // when card is selected.  
+            // Can insert here but would be inconsistent with TeachSession behavior
+            /* 
+            const buttonSubmit = activity.channelData && activity.channelData.imback
+            if (this.state.selectedActivity && buttonSubmit) {
                 await this.props.onInsertInput(this.state.currentTrainDialog!, this.state.selectedActivity, userInput.text, this.state.addUserInputSelectionType)
             }
+            */
             // If next action should be score, need to insert, not continue
-            else if (this.waitingForScore()) {
+            if (this.waitingForScore()) {
```

Comment:
This feels super hacky, but webchat doesn't differentiate between button clicks and clicking on an action when posting a select activity call, so I have to track by noting that the previous input was an imback